spot_driver/spot_driver/launch/spot_launch_helpers.py

```diff
@@ -2,12 +2,51 @@
 
 import logging
 import os
-from typing import Optional, Tuple
+from typing import Any, Dict, List, Optional, Tuple
 
 import yaml
 
 from spot_wrapper.wrapper import SpotWrapper
 
+COLOR_END = "\33[0m"
+COLOR_YELLOW = "\33[33m"
+
+
+def get_ros_param_dict(config_file_path: str) -> Dict[str, Any]:
+    """Get a dictionary of parameter_name: parameter_value from a ROS config yaml file.
+
+    Args:
+        config_file_path (str): Path to the config yaml.
+
+    Raises:
+        FileNotFoundError: If the config file doesn't exist
+        YamlError: If the yaml cannot be parsed
+        ValueError: If the yaml file doesn't follow ROS conventions
+
+    Returns:
+        dict[str, Any]: dictionary of parameter_name: parameter_value.
+        If there is no config file, returns the empty dictionary.
+    """
+    # an empty config file path is a valid way to start the driver, so here we just return the empty dictionary.
+    if not config_file_path:
+        return {}
+    if os.path.isfile(config_file_path):
```

Comment:
@khughes-bdai nit: file existence != file availability, `open` may still fail on bad permissions. All that to say, you could just let `open` raise if need be.